### PR TITLE
[ENH] Uniform Continuous distribution

### PR DESCRIPTION
**Pull request by an20805**
#### What does this implement/fix? Explain your changes.

Implemented Uniform Continuous Probability Distribution, towards https://github.com/sktime/skpro/issues/22

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### What should a reviewer concentrate their feedback on?
Testing the distribution

#### Did you add any tests for the change?
No

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/skpro/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/skpro/blob/main/.all-contributorsrc) in the `skpro` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [X] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Comment by fkiraly** [comment]
you said you had some questions in 22 - what do you need help with?

**Comment by an20805** [comment]
Its showing Read the Docs build failed. How can it be solved?

**Comment by fkiraly** [comment]
> Its showing Read the Docs build failed. How can it be solved?

That look new - a syntax error in one of the json files related to docs.

**Comment by fkiraly** [comment]
I am suspecting `switcher.json`, but why would that start breaking now?

File: skpro/distributions/uniform.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    _tags = {
+        "authors": ["an20805"],  
+        "capabilities:approx": ["pdfnorm"],
+        "capabilities:exact": ["pdf", "log_pdf", "cdf", "ppf", "mean", "var"],
+        "distr:measuretype": "continuous",
+    }
+
+    def __init__(self, lower, upper, index=None, columns=None):
+        if lower >= upper:
+            raise ValueError("Upper bound must be greater than lower bound.")
```
Comment (VascoSch92) [inline skpro/distributions/uniform.py:40]:
```suggestion
            raise ValueError(f"Upper bound ({upper}) must be greater than lower bound {lower}.")
```
in this way the error message gives more information about the problem

**Review by fkiraly — APPROVED** [review]
updated to new interface contract introduced in https://github.com/sktime/skpro/pull/265

**Comment by fkiraly** [comment]
failures are due to bugs fixed in https://github.com/sktime/skpro/pull/281, that should be merged first and tests re-run

**Pull request merged**